inspectors/utils/doubleinputvalidator.cpp

```diff
@@ -1,56 +1,107 @@
 #include "doubleinputvalidator.h"
 
-DoubleInputValidator::DoubleInputValidator(QObject *parent) : QValidator(parent)
-      {
-      }
+DoubleInputValidator::DoubleInputValidator(QObject* parent) : QValidator(parent)
+{
+}
 
 void DoubleInputValidator::fixup(QString& string) const
-      {
-      if (!string.contains("."))
-            string.append(".00");
+{
+    auto zeros = [](int num)->QString {
+        QString s = QString();
+        for (int i = 0; i < num; i++)
+            s.append("0");
+        return s;
+    };
 
-      if (string.startsWith("."))
-            string.prepend("0");
+    if (!string.contains("."))
+        string.append("." + zeros(m_decimal));
 
-      if (string.endsWith("."))
-            string.append("00");
+    if (string.startsWith("."))
+        string.prepend("0");
 
-      QStringList strList = string.split(".", QString::SkipEmptyParts);
+    if (string.endsWith("."))
+        string.append(zeros(m_decimal));
 
-      QString intPart = strList.at(0);
-      QString floatPart = strList.at(1);
+    QStringList strList = string.split(".", QString::SkipEmptyParts);
 
-      if (intPart == QString("-"))
-            intPart.append("0");
-      if (floatPart.length() == 1)
-            floatPart.append("0");
+    QString intPart = strList.at(0);
+    QString floatPart = strList.at(1);
 
-      if (intPart.contains(QRegExp("^0{1,3}$")))
-            intPart = QString("0");
-      else if (intPart.contains(QRegExp("^\\-0{1,3}$")))
-            intPart = QString("-0");
+    if (intPart == QString("-"))
+        intPart.append("0");
+    if (floatPart.length() < m_decimal)
+        floatPart.append(zeros(m_decimal - floatPart.length()));
 
-      if (intPart == QString("-0") && floatPart == QString("00"))
-            intPart = QString("0");
+    if (intPart.contains(QRegExp("^0{1,3}$")))
+        intPart = QString("0");
+    else if (intPart.contains(QRegExp("^\\-0{1,3}$")))
```

Comment:
there is a feeling that these two conditions are echoing each other

`if (intPart.contains(QRegExp("^\\-0{1,3}$")))`
`if (intPart == QString("-"))`

it seems it can be reduced to one
`intPart.contains(QRegExp("^\\-0{0,3}$"))`